### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by littlewhitecloud**
Something goes wrong in #437 so I open a new pull request, sorry.
- [x] make self-hosted compiler output the same results
- [x] add more tests

### CODE REVIEW

**Comment by littlewhitecloud** [comment]
- [ ] tokenize

File: src/typecheck.c [hunk 1]
```diff
@@ -87,9 +87,8 @@ static const char *short_type_description(const Type *t)
         case TYPE_ARRAY:
             return "an array type";
         case TYPE_BOOL:
-            // TODO: Is it possible to get this in an error message?
-            return "the built-in bool type";
-    }
+            assert(0);
```
Comment (Akuli) [inline src/typecheck.c:90]: It is actually possible to get this in an error message:
```
def foo() -> None:
    x = True.asdf()
```

Comment (littlewhitecloud) [reply src/typecheck.c:90]: Fine, I never thought user may run the code like this.

**Comment by littlewhitecloud** [comment]
Find a lot of todos in each file and add tests for them if they said they need tests

File: src/typecheck.c [hunk 2]
```diff
@@ -246,8 +245,6 @@ static Signature handle_signature(FileTypes *ft, const AstSignature *astsig, con
     else
         sig.returntype = type_from_ast(ft, &astsig->returntype);
 
-    // TODO: validate main() parameters
-    // TODO: test main() taking parameters
```
Comment (Akuli) [inline src/typecheck.c:250]: We could keep these todo comments until #257 is fixed.

Comment (littlewhitecloud) [reply src/typecheck.c:250]: Really? The self-hosted compiler can get args now

Comment (Akuli) [reply src/typecheck.c:250]: Good point. We should still add validation: if main has parameters, there must be 2 of them, of types `int` and `byte**`.

**Review by Akuli — APPROVED** [review]
Thanks!

**Pull request merged**